### DISCUSSION TITLE
authenticityToken - how to clear it out

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1434] [General]
I tried a similar question on the F&A board but got no response so I'll try here and hope for more eyes.

How do others clear out the authenticityToken from where it is stored? In our app, the logout seems to do it and it allows login just fine. But when the browser times out it does not clear out and I've tried removing a cookie by that name, I've tried destroying the session scope in case it's in there, but first login attempt after timeout, it throws the invalidAuthenticityToken error. But that clears it out and the 2nd attempt at login succeeds. I don't quite understand what is going on but any tips on how others do it would be appreciated.

**Comment by cfcoder** [comment]
No one has responded to this yet but I will add another observation I've made about it. I found that the authenticity token remains in the login form that gets left after the auto-logout we have implemented but then after some time (usually overnight) it becomes stale. Straight away filling in that login form and submitting carries the stale authenticity token with it and fails the validation. Second try of course works since a reload brings in a new authenticity token.
So my question is basically how do others handle this situation?  Is there a way to replace the token as part of the submit or is it best to just auto-logout to a logout page and leave it there with a button that will take the user to the login assuming they will do it straight away and which will then come with a fresh token?

**Comment by bpamiri** [comment]
Bruce,

If it were me, I would auto logout to a separate screen that informs the user that their session has been closed out and they have been logged out. I’ve seen this UX with some banking websites. 

--Peter Amiri
  ***@***.***
